business/iter8.go

```diff
@@ -241,3 +289,90 @@ func (in *Iter8Service) GetAnalyticPort() int {
 	}
 	return analyticConfig.Port
 }
+
+func (in *Iter8Service) GetIter8RoutingReferences(namespace string, servicename string) (routingReferences []models.RoutingReference, err error) {
+	promtimer := internalmetrics.GetGoFunctionMetric("business", "Iter8Service", "GetIter8Metrics")
+	defer promtimer.ObserveNow(&err)
+	Gateways := &models.Gateways{}
+	gws, err := in.k8s.GetGateways(namespace)
+	VirtualServices := &models.VirtualServices{Items: []models.VirtualService{}}
+	routingReferences = make([]models.RoutingReference, 0)
+	gwNames := make([]string, 0)
+	if err != nil {
+		return routingReferences, err
+	}
+
+	Gateways.Parse(gws)
+	for _, gw := range gws {
+		gwNames = append(gwNames, gw.GetObjectMeta().Name)
+	}
+	vs, err := in.k8s.GetVirtualServices(namespace, servicename)
```

Comment:
I think this can be done in a better way.
I mean, here the Iter8Service is duplicating basically a feature that is dont in the IstioConfigService.
As the Iter8Service has access to the Layer object, it can access to that service and query the VirtualServices or others objects getting advantage of the caching and parsing.

Then the Iter8Service should only consume that service.

Let me know if it makes sense, I can also help to propose a PR if perhaps there is some difficulty but I guess if you check the "Layer" type probably you will get examples about what you want to do.

Reply:
yes, it makes sense. I just didn't realize the existing capabilities.

Reply:
@lucasponce I am replacing the GetGateway and GetVirtualServices with the following code section. Is this what you have in mind? 
```
istioCfg, err := in.businessLayer.IstioConfig.GetIstioConfigList(IstioConfigCriteria{
		IncludeGateways : true,
		IncludeVirtualServices:  true,
		Namespace:               namespace,
	})
```

Reply:
@jadeyliu939 yes, using the IstioConfig API enables the caching.
Thanks !